sale_product_approval/models/product_state.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2021 Open Source Integrators
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ProductState(models.Model):
+    _inherit = "product.state"
+
+    approved_sale = fields.Boolean(
+        string="Approved to be Sold",
```

Comment:
To not break current behavior once installed 
```suggestion
        string="Approved to be Sold",
        default=True,
```